### ISSUE TITLE
[Bug] Password unlock enforced on re-open

### ISSUE THREAD

**Issue by KaKi87** [wireapp/wire-android#2485]
**How to reproduce :**

In order to reproduce this issue, Android fingerprint unlock must be enabled.
At the beginning, you're focused on your launcher or another app than Wire.

- Open wire
- Do not unlock it
- Go back to launcher or another app
- Open wire again

**Expected behavior :**

The app should be unlocked

**Actual behavior :**

The app will ask for password despite successfull fingerprint unlock

**Device info :**
- Wire : 3.43.870
- Android OS : 10
- Firmware : stock
- ROM : Havoc
- Mobile phone model/manufacturer : Xiaomi Mi 8 "dipper"
- Mobile network type (EDGE/LTE/Wi-Fi/Offline) : LTE

**Additional context :**

This issue should be adapted with #2483 if it's ever resolved.

**Comment by KaKi87** [comment]
Hello,

Any news on this ?

Thanks

**Comment by android10** [comment]
@KaKi87 first of all, thanks for reporting in such a detailed way. Totally appreciated.  

**I have a question for you: have you tried on the stock/official ROM?**

Cause it seems you are using a custom rom, which might be the problem, especially since they are modified internally. Also a logcat output could be very helpful in order to decide support.

**Comment by android10** [comment]
@KaKi87 custom ROMS modify internals in terms of code in order to optimize and gain performance. There are tons of those out there, which put as in a position that we cannot test everything on every existent customized android version. 

That is why if this cannot be reproduced on the official/supported android version, there is not much we can do on our side, unless that the community help us by sending logs and more information. 

We do our best to fix everything but we also need collaboration/contribution and this is not really helping:

>@android10 And you, did you tried ?
Also, how do you prove that using custom ROMs cause bugs that doesn't happen on proprietary ones ? Sorry I do not accept that answer.

I would expect that instead of answering one question with another question, that we collaborate together. That is the open source code of conduct.

If you have another device out there (with another rom), I will appreciate further information (specially if this is failing) so we can prioritize and add this to the backlog. Keep in mind if that is something specific only for the customized rom you are using, then probably the problem comes from that specific android rom. 

We definitely want to help and, one more time, you did a great job describing the issue. Thanks for reporting.

**Comment by makingthematrix** [comment]
Hi,

I think this is fixed already. The bug was that we created new dialogs on top of old ones. When the user goes out of the app and back, a new password dialog was created, and a new biometric prompt attached to it, but after authentication the old dialog, the one from the previous opening of the app, was still there. It should have been dismissed the moment the user goes out of the app.
Fixed in https://github.com/wireapp/wire-android/pull/2543
